### PR TITLE
Feature/dynamic analysis attributes

### PR DESCRIPTION
**Pull request by blabadi**
- add AnalysisTypeId
- add data node to catch all extra data
- create index with alias on startup
- add data node to send all extra data to elastic
- fix tests and update fixtures
- update documentation

### CODE REVIEW

File: maestro-app/src/main/resources/config/application.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ maestro:
       organization: ICGC
       country: CA
     # you can other SONGs as needed
-    - code: song.overture
+    - code: song.overture1
```
Comment (Buwujiu) [inline maestro-app/src/main/resources/config/application.yml]: Can you explain why code is changed to `song.overture1`?

Comment (blabadi) [reply maestro-app/src/main/resources/config/application.yml]: typo will fix it

File: maestro-domain/src/main/java/bio/overture/maestro/domain/entities/indexing/FileCentricAnalysis.java [hunk 2]
```diff
@@ -34,11 +37,36 @@
...
     private Map<String, Object> experiment;
+
+    /**
+     * this field is to capture the dynamic fields in the analysis.
+     * it's the responsibility of the users to make sure the mapping is consistent with
+     * the different fields that they want to add/index, they are also responsibile
+     * to add the mappings of these fields or reindex appropriately.
+     */
+    @NonNull
+    private final Map<String, Object> data = new TreeMap<>();
```
[the remark this answers is not in the document]
Comment (blabadi) [reply maestro-domain/src/main/java/bio/overture/maestro/domain/entities/indexing/FileCentricAnalysis.java]: no specific reason, I copied it from SONGs definition, but a HashMap should work, I don't believe it needs to be a TreeMap there too as no data structure specific features needed here, but probably Rob can confirm

File: maestro-app/src/main/resources/config/application.yml [hunk 3]
```diff
@@ -39,8 +39,8 @@ maestro:
   # List of Genomic files repositories (SONGs)
   repositories:
     # these properties will be used in the document (see ../file_centric.json)
-    - code: song.overture # must be unique & must match song.serverId if using kafka integration with song
-      url: http://localhost:8080
+    - code: song.qa.argo # must be unique & must match song.serverId if using kafka integration with song
+      url: https://song.qa.argo.cancercollaboratory.org/
```
[the remark this answers is not in the document]
Comment (blabadi) [reply maestro-app/src/main/resources/config/application.yml]: will revert

File: maestro-app/src/main/resources/config/application.yml (hunk 1 above)
Comment (blabadi) [reply maestro-app/src/main/resources/config/application.yml]: actually it is not a typo, i changed the first entry as Dusan mentioned in his comment so it seems correct to keep.

**Pull request merged**